task/task-pangolin-ropsten/src/service/ropsten/ropsten_logs_handler.rs

```diff
@@ -156,22 +173,80 @@ fn build_txs(
 }
 
 impl RopstenLogsHandler {
+    async fn is_verified(&self, tx: &EthereumTransaction) -> anyhow::Result<bool> {
+        Ok(self
+           .darwinia_client
+           .verified(tx.block_hash, tx.index)
+           .await?
+           || self
+           .darwinia_client
+           .verified_issuing(tx.block_hash, tx.index)
+           .await?
+          )
+    }
+
+    fn is_redeem_submitted(&self, tx: &EthereumTransaction) -> bool {
+        let txset = self.waited_redeem.get(&tx.block);
+        if let Some(txs) = txset {
+            return txs.contains(&tx)
+        }
+        false
+    }
+
+    async fn check_redeem(&mut self, check_position: u64) -> anyhow::Result<()> {
```

Comment:
why not index with <block_number, tx_id>? 

@fewensa please help review, can we simplify this implementation.